### PR TITLE
[vcpkg baseline][libharu] update to 2.4.0-rc1

### LINKED ISSUE #25537: [libharu] build failure - File does not have the expected hash
**Host Environment**
- OS: windows 10
- Compiler: MSVC 19.29.30145.0
-    vcpkg-tool version: 2022-06-17-9268e366206712e38102b28dbd1617697a99ff2e
    vcpkg-scripts version: 4826ede84 2022-07-01 (20 hours ago)

**To Reproduce**
1.vcpkg install qt5:x64-windows
2.vcpkg install vtk[qt]:x64-windows --recurse
ps. failed at step 2, not bath of them

**Failure logs**
C:\Users\snowm\code\vcpkg>vcpkg install vtk[qt]:x64-windows --recurse
Computing installation plan...
The following packages will be built and installed:
  * libharu[core,notiffsymbols]:x64-windows -> 2017-08-15#10
  * libiconv[core]:x64-windows -> 1.17
  * libogg[core]:x64-windows -> 1.3.5
  * libtheora[core]:x64-windows -> 1.2.0alpha1-20170719#3
  * libxml2[core]:x64-windows -> 2.9.14
  * lz4[core]:x64-windows -> 1.9.3#4
  * netcdf-c[core,dap,hdf5,nczarr,netcdf-4,platform-default-features]:x64-windows -> 4.8.1#2
  * nlohmann-json[core]:x64-windows -> 3.10.5#3
  * pegtl-2[core]:x64-windows -> 2.8.3#1
  * proj[core,net,tiff]:x64-windows -> 9.0.1
  * pugixml[core]:x64-windows -> 1.12.1
  * qt5-activeqt[core]:x64-windows -> 5.15.5
  * qt5-base[core]:x64-windows -> 5.15.5#1
  * qt5-declarative[core]:x64-windows -> 5.15.5
  * qt5-imageformats[core]:x64-windows -> 5.15.5#1
  * qt5-svg[core]:x64-windows -> 5.15.5
  * qt5-tools[core]:x64-windows -> 5.15.5
  * qt5-xmlpatterns[core]:x64-windows -> 5.15.5
  * sqlite3[core,tool]:x64-windows -> 3.37.2#2
  * utfcpp[core]:x64-windows -> 3.2.1
    vtk[core,qt]:x64-windows -> 9.0.3-pv5.9.1#10
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet "x64-windows"...
Restored 0 packages from C:\Users\snowm\AppData\Local\vcpkg\archives in 803 us. Use --debug to see more details.
Installing 1/21 libharu:x64-windows...
Building libharu[core,notiffsymbols]:x64-windows...
-- Downloading https://github.com/libharu/libharu/pull/157.diff -> libharu-shading-pr-157.patch...
[DEBUG] To include the environment variables in debug output, pass --debug-env
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'manifests' = off
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Downloading https://github.com/libharu/libharu/pull/157.diff
Error: Failed to download from mirror set:
File does not have the expected hash:
             url : [ https://github.com/libharu/libharu/pull/157.diff ]
       File path : [ C:\Users\snowm\code\vcpkg\downloads\libharu-shading-pr-157.patch.14644.part ]
   Expected hash : [ f2ddb22b54b4eccc79400b6a4b2d245a221898f75456a5a559523eab7a523a87dfc5dfd0ec5fb17a771697e03c7ea6ed4c6095eff73e0a4302cd6eb24584c957 ]
     Actual hash : [ 10562c8c9c6975b74d625912e7d3b850376941d5e46136679371ce1315f416b6b8e91d73fd1257b907327b2d71d50b8c026547203f280e42fb7b1901245de03c ]

[DEBUG] D:\a\_work\1\s\src\vcpkg\base\downloads.cpp(788):
[DEBUG] Time in subprocesses: 0 us
[DEBUG] Time in parsing JSON: 2 us
[DEBUG] Time in JSON reader: 0 us
[DEBUG] Time in filesystem: 1943 us
[DEBUG] Time in loading ports: 0 us
[DEBUG] Exiting after 1.245 s (1234976 us)

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:24 (message):

      Failed to download file with error: 1
      If you use a proxy, please check your proxy setting. Possible causes are:

      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.

      2. You are using Fiddler. Currently a bug (https://github.com/microsoft/vcpkg/issues/17752)
         will set HTTPS_PROXY to `https://fiddler_address:port` which lead to problem 1 above.
         Workaround is open Windows 10 Settings App, and search for Proxy Configuration page,
         Change `http=address:port;https=address:port` to `address`, and fill the port number.

      3. Your proxy's remote server is out of service.

      In future vcpkg releases, if you are using Windows, you no longer need to set
      HTTP(S)_PROXY environment variables. Vcpkg will simply apply Windows IE Proxy
      Settings set by your proxy software. See (https://github.com/microsoft/vcpkg-tool/pull/49)
      and (https://github.com/microsoft/vcpkg-tool/pull/77)

      Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:249 (z_vcpkg_download_distfile_show_proxy_and_fail)
  ports/libharu/portfile.cmake:6 (vcpkg_download_distfile)
  scripts/ports.cmake:146 (include)

error: building libharu:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+libharu
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[libharu]+Build+error
Include '[libharu] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-06-17-9268e366206712e38102b28dbd1617697a99ff2e
    vcpkg-scripts version: 4826ede84 2022-07-01 (20 hours ago)

### PR DESCRIPTION
**Pull request by Adela0814**
Fix https://github.com/microsoft/vcpkg/issues/25537

The duplicated symbols of `tiff/libharu` have been fixed, so remove the patch `tiff.patch` and the feature.

### CODE REVIEW

**Comment by larshg** [comment]
Missing updating [here](https://github.com/microsoft/vcpkg/blob/af9fe3e266d94c45885d2252d0f8f805a4c4dd41/ports/vtk/vcpkg.json#L19-L24) - thx for the PR 😄

**Comment by larshg** [comment]
Seems libharu doesn't produce any .lib files, so VTK fails 😢 

```
-- Could NOT find LibHaru (missing: LibHaru_LIBRARY) (found suitable version "2.4.0-dev", minimum required is "2.4.0")
CMake Error at CMake/vtkModule.cmake:4228 (message):
  Could not find the LibHaru external dependency.
Call Stack (most recent call first):
  CMake/vtkModule.cmake:4784 (vtk_module_find_package)
  CMake/vtkModule.cmake:4655 (vtk_module_third_party_external)
  ThirdParty/libharu/CMakeLists.txt:1 (vtk_module_third_party)

```

**Comment by Adela0814** [comment]
> Missing updating [here](https://github.com/microsoft/vcpkg/blob/af9fe3e266d94c45885d2252d0f8f805a4c4dd41/ports/vtk/vcpkg.json#L19-L24) - thx for the PR 😄

Thanks for your reminder, I'm working on this.

**Comment by Adela0814** [comment]
> Seems libharu doesn't produce any .lib files, so VTK fails on windows-x64 😢
> 
> ```
> -- Could NOT find LibHaru (missing: LibHaru_LIBRARY) (found suitable version "2.4.0-dev", minimum required is "2.4.0")
> CMake Error at CMake/vtkModule.cmake:4228 (message):
>   Could not find the LibHaru external dependency.
> Call Stack (most recent call first):
>   CMake/vtkModule.cmake:4784 (vtk_module_find_package)
>   CMake/vtkModule.cmake:4655 (vtk_module_third_party_external)
>   ThirdParty/libharu/CMakeLists.txt:1 (vtk_module_third_party)
> ```

Can you provide the repro steps, I can build successfully.
image

**Comment by larshg** [comment]
This looks odd: https://github.com/libharu/libharu/blob/6997cf775b2345e3db82ac774fe2931faf348458/src/CMakeLists.txt#L90-L95 - there are no targets in the install command. Seems to be fixed on master.

**Comment by larshg** [comment]
Still no import libraries (.lib):
image

**Comment by dg0yt** [comment]
No exported symbols?

**Comment by Adela0814** [comment]
> Still no import libraries (.lib): image

I'm fixing this, I will teel you if ready.

**Comment by larshg** [comment]
Got .lib now 😄 - but VTK still fails to find libharu:
```
-- Could NOT find LibHaru (missing: LibHaru_LIBRARY) (found suitable version "2.4.0-dev", minimum required is "2.4.0")
CMake Error at CMake/vtkModule.cmake:4228 (message):
  Could not find the LibHaru external dependency.
Call Stack (most recent call first):
  CMake/vtkModule.cmake:4784 (vtk_module_find_package)
  CMake/vtkModule.cmake:4655 (vtk_module_third_party_external)
  ThirdParty/libharu/CMakeLists.txt:1 (vtk_module_third_party)
```

**Comment by Adela0814** [comment]
To do:  update `vtk` port  (fix FindLibHaru.cmake).

**Comment by larshg** [comment]
> @larshg Please try the latest commit.

Already did and it build succesfully 👍

File: ports/vtk/FindLibHaru.patch [hunk 1]
```diff
@@ -0,0 +1,98 @@
+diff --git a/CMake/FindLibHaru.cmake b/CMake/FindLibHaru.cmake
+index 1f5b881f..1655ce25 100644
+--- a/CMake/FindLibHaru.cmake
++++ b/CMake/FindLibHaru.cmake
+@@ -1,87 +1,6 @@
+-find_path(LibHaru_INCLUDE_DIR
```
Comment (JackBoosY) [inline ports/vtk/FindLibHaru.patch:6]: Please use `if (0) ... endif()` to reduce the patch size.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/vtk/FindLibHaru.patch [hunk 2]
```diff
@@ -0,0 +1,21 @@
+diff --git a/CMake/FindLibHaru.cmake b/CMake/FindLibHaru.cmake
+index 1f5b881..8a4c426 100644
+--- a/CMake/FindLibHaru.cmake
++++ b/CMake/FindLibHaru.cmake
+@@ -1,3 +1,11 @@
++find_package(unofficial-libharu CONFIG REQUIRED)
++set(LibHaru_FOUND 1)
```
[the remark this answers is not in the document]
Comment (Adela0814) [reply ports/vtk/FindLibHaru.patch:7]: The build fails once I remove the `set(LibHaru_FOUND 1)`, This is the problem of `vtk`, I actually don't recommend modifying too much `vtk` related in this PR. I tried to fix it, but unfortunately `vtk` is not easy to fix.

[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/vtk/FindLibHaru.patch:7]: `find_package(LibHaru CONFIG NAMES unofficial-libharu REQUIRED)` must set `LibHaru_FOUND` - if it is found.

[the remark this answers is not in the document]
Comment (Adela0814) [reply ports/vtk/FindLibHaru.patch:7]: Yes, it is expected, but in fact the following error will appear, and the function of vtk needs to be fixed.

```
CMake Error at E:/vcpkg/scripts/buildsystems/vcpkg.cmake:824 (_find_package):
  Could not find a configuration file for package "LibHaru" that is
  compatible with requested version "2.4.0".
 The following configuration files were considered but not accepted:
   E:/vcpkg/installed/x64-windows/share/unofficial-libharu/unofficial-libharu-config.cmake, version: unknown
Call Stack (most recent call first):
  CMake/FindLibHaru.cmake:1 (find_package)
  E:/vcpkg/scripts/buildsystems/vcpkg.cmake:824 (_find_package)
  CMake/vtkModule.cmake:4222 (find_package)
```

[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/vtk/FindLibHaru.patch]: Wait, you patch `CMake/FindLibHaru.cmake`... normally we would patch the location of `find_package(LibHaru)`, making the find module obsolete.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**